app/src/main/java/com/gathering/android/event/home/EventLocationComparator.kt

```diff
@@ -0,0 +1,20 @@
+package com.gathering.android.event.home
+
+import android.location.Location
+import com.gathering.android.event.home.model.Event
+import javax.inject.Inject
+
+
+class EventLocationComparator @Inject constructor() : Comparator<Event> {
+    override fun compare(event1: Event, event2: Event): Int {
+        val distA = event1.location?.distanceTo(getCurrentLocation())
+        val distB = event2.location?.distanceTo(getCurrentLocation())
+
+        return (distA ?: 0f).compareTo(distB ?: 0f)
+    }
+
+    private fun getCurrentLocation(): Location {
```

Comment:
take this method to a new class called `LocationProvider` and use Android framework `fusedLocationClient.lastLocsation` to get the last known user location. (look [here](https://developer.android.com/training/location/retrieve-current))

you will need to ask for permissions for that !

Reply:
it will be addressed in T29